src/patterns/validation/index.md.njk

```diff
@@ -0,0 +1,92 @@
+
+There are separate patterns for:
+
++ [‘there is a problem with the service’ pages](/patterns/problem-with-the-service-pages/)
++ [‘page not found’ pages](/patterns/page-not-found-pages/)
++ [‘service unavailable’ pages](/patterns/service-unavailable-pages/)
+
+## How it works
+
+Validation should refuse to accept:
+
++ information that can’t be correct
++ information that’s too ambiguous for you to use
++ missing information, if the information is required
+
+For example, if you’re asking for someone’s date of birth you shouldn’t accept ‘13’ in the month field.
+
+### How to tell the user about validation errors
+
+If the user's answers fail validation:
+
++ show them the page again, with the form fields as the user filled them in
++ add ‘Error: ’ to the beginning of the page `<title>` so screen readers read it out as soon as possible
++ show an [error summary](/components/error-summary/) at the top of the page
++ show [error messages](/components/error-message/) next to fields with errors
+
+Read guidance on [writing good error messages](/components/error-message#be-clear-and-concise).
+
+### When to tell the user about validation errors
+
+Do not validate when the user moves away from a field. Wait until they try to move to the next part of the service - usually by clicking the ‘continue’ or ‘submit’ button at the bottom of the page.
+
+Generally speaking, avoid validating the information in a field before the user has finished entering it. This sort of validation can cause problems - especially for users who type more slowly.
+
+Only add this sort of validation if your user research shows that, on balance, it solves more problems for users than it causes. For example, the [character count component](/components/character-count/) shows users an error message when they go over the character limit. Because it’s important that users do not spend time and effort writing out a response that turns out to be too long.
+
+If you do use this sort of validation, [make sure you do it in a way that’s accessible](https://www.gov.uk/service-manual/technology/accessibility-for-developers-an-introduction).
+
+## Client side and server side validation
+
+There are 2 ways to carry out validation:
+
++ server side validation
++ client side validation (in the browser, using either JavaScript or HTML)
+
+You’ll always need to carry out server side validation, because there’s no guarantee that client side validation will work in all circumstances. For example users can bypass client side validation, or [JavaScript can fail to load](https://www.gov.uk/service-manual/technology/using-progressive-enhancement).
+
+Avoid adding client side validation unless you can establish that there’s a need for it. For example, because user research shows it’s helpful to users if you validate the information they’re providing as they type.
+
+If you do use client side validation, make sure your client side and server side validation rules do not conflict.
+
```

Comment:
Is it worth mentioning some of the common pitfalls of client side validation?

For instance:
* Distracting / jarring / trigger to easily
* Reflows the page

Reply:
We don't want to get too much into the technical aspects of client side validation, as we're not recommending it. If teams find a need to add it, they will have to look into accessibility, reliability etc for themselves. If common user needs are found, we could document tried and tested approaches.